app/controllers/decidim/pages_controller.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This controller serves static pages using HighVoltage.
+  class PagesController < Decidim::ApplicationController
+    layout "layouts/decidim/application"
+
+    helper_method :page, :pages, :page_content_blocks
+    helper CtaButtonHelper
+    helper Decidim::SanitizeHelper
+
+    before_action :legacy_redirect, :set_default_request_format, :accessibility_redirect
+
+    def index
+      enforce_permission_to :read, :public_page
+      @topics = StaticPageTopic.where(organization: current_organization)
+      @orphan_pages = StaticPage.where(topic: nil, organization: current_organization)
+    end
+
+    def show
+      @page = current_organization.static_pages.find_by!(slug: params[:id])
+      enforce_permission_to :read, :public_page, page: @page
+      @topic = @page.topic
+      @pages = @topic&.pages
+      raise
+    end
+
+    private
+
+    def legacy_redirect
+      redirect_to decidim.page_path("terms-of-service") if params[:id] == "terms-and-conditions"
+    end
+
+    def accessibility_redirect
+      if params[:id] == "accessibilitat"
+        if params[:locale] == "es"
+          redirect_to "https://ajuntament.barcelona.cat/declaracio-accessibilitat/es/decidim-barcelona"
+        else
+          redirect_to "https://ajuntament.barcelona.cat/declaracio-accessibilitat/ca/decidim-barcelona"
+        end
+      end
+    end
```

Comment:
Performing it on the `routes.rb` file could be tricky. So, instead of this hard monkey-patch, you could do an override file with a concern only including the new `before_action` and the method that could be refactored as below:

```suggestion
    def accessibility_redirect
      return if params[:id] != "accessibilitat"

      redirect_to "https://ajuntament.barcelona.cat/declaracio-accessibilitat/#{I18n.locale}/decidim-barcelona"
    end
```